lib/PreviewProvider.js

```diff
@@ -342,18 +298,16 @@ PreviewProvider.prototype = {
         let responseJson = yield response.json();
         this._tabTracker.handlePerformanceEvent(event, "embedlyProxyRequestReceivedCount", responseJson.urls.length);
         this._tabTracker.handlePerformanceEvent(event, "embedlyProxyRequestSucess", 1);
-        let currentTime = Date.now();
-        newLinks.forEach(link => {
+
+        for (let link of newLinks) {
           let data = responseJson.urls[link.sanitized_url];
           if (!data) {
             return;
           }
-          ss.storage.embedlyData[link.cache_key] = Object.assign({}, ss.storage.embedlyData[link.cache_key], data);
-          if (updateAccessTime) {
-            ss.storage.embedlyData[link.cache_key].accessTime = currentTime;
-          }
-          ss.storage.embedlyData[link.cache_key].refreshTime = currentTime;
-        });
+          // Save the entry in the metadata database
+          let entry = Object.assign({}, link, {expired_at: (this.options.metadataTTL) + Date.now()});
+          yield this._metadataStore.asyncInsert([Object.assign({}, data, entry)]);
```

Comment:
We don't need to yield here, instead, as @sarracini said, we should catch the exceptions here. Unique key violation is one of the possible exceptions. Other exceptions might also be raised if the SQLite runs into other issues, e.g. out of disk space, oom, transaction timeout etc.